src/util/commit-exclude.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Commit} from '../commit';
+import {ReleaserConfig} from '../manifest';
+import {normalizePaths} from './commit-utils';
+
+export type CommitExcludeConfig = Pick<ReleaserConfig, 'excludePaths'>;
+
+export class CommitExclude {
+  private excludePaths: Record<string, string[]> = {};
+
+  constructor(config: Record<string, CommitExcludeConfig>) {
+    Object.entries(config).forEach(([path, releaseConfig]) => {
+      if (releaseConfig.excludePaths) {
+        this.excludePaths[path] = normalizePaths(releaseConfig.excludePaths);
+      }
+    });
+  }
+
+  excludeCommits<T extends Commit>(
+    commitsPerPath: Record<string, T[]>
+  ): Record<string, T[]> {
+    const filteredCommitsPerPath: Record<string, T[]> = {};
+    Object.entries(commitsPerPath).forEach(([path, commits]) => {
+      if (this.excludePaths[path]) {
+        commits = commits.filter(commit =>
+          this.shouldInclude(commit, this.excludePaths[path])
+        );
+      }
+      filteredCommitsPerPath[path] = commits;
+    });
+    return filteredCommitsPerPath;
+  }
+
+  private shouldInclude(commit: Commit, excludePaths: string[]): boolean {
+    return (
+      !commit.files ||
+      !commit.files.every(file =>
+        excludePaths.some(path => file.indexOf(`${path}/`) === 0)
+      )
```

Comment:
I think there's an edge case here that will be difficult to account for if you split the exclude from the include logic. The original `CommitSplit` class does not strip files out that don't match the path -- it's just used to determine if that commit touches that directory (it can touch multiple components).

This looks like it's checking if there's any file that does not match the exclude path pattern. Consider a commit that touches: 
* `a/b/c`
* `d/e/f`
* `d/e/g`

We have 2 components with paths `a` and `d` and `d` is configured to exclude `d/e/*`. This commit has a file (`a/b/c`) that does not match the exclude pattern, and will not be excluded. If I understand the feature correctly, this commit should be excluded (there's no commit under path `d` that does not include `d/e/*`).

Reply:
hm, true there is such case. 
Main reason for splitting this into two (aside of not adding additional logic tree to commit split) was the need to handle exclusion on `.` root scope
Thanks for catching up that usecase! Will update soon with changes

Reply:
I spent some time on adding the logic into commit split, but in the end decided to handle exclusion logic separately.
I have updated it to handle exclusion only based on relevant to project files.

Main reasoning of keeping it as "split" then "filter" is logic is a bit cleaner that way
1. no special case to iterate over `.`
2. commit split is not bloated with additional cases to handle and does only 1 thing, it splits commits per package

What do you think?
All my attempts to include that logic in commit-split itself did not look pretty, unfortunately.

Reply:
@chingor13 i think now it covers all needed usecases. Could you take a look when you have time?